sdk-tests/src/test/java/io/dapr/it/pubsub/stream/PubSubStreamIT.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 The Dapr Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package io.dapr.it.pubsub.stream;
+
+import io.dapr.client.DaprClient;
+import io.dapr.client.DaprPreviewClient;
+import io.dapr.client.SubscriptionListener;
+import io.dapr.client.domain.CloudEvent;
+import io.dapr.it.BaseIT;
+import io.dapr.it.DaprRun;
+import io.dapr.utils.TypeRef;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static io.dapr.it.Retry.callWithRetry;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
+public class PubSubStreamIT extends BaseIT {
+
+  private static final int NUM_MESSAGES = 10;
+  private static final String TOPIC_NAME = "stream-topic";
+  private static final String PUBSUB_NAME = "messagebus";
+
+  private final List<DaprRun> runs = new ArrayList<>();
+
+  private DaprRun closeLater(DaprRun run) {
+    this.runs.add(run);
+    return run;
+  }
+
+  @AfterEach
+  public void tearDown() throws Exception {
+    for (DaprRun run : runs) {
+      run.stop();
+    }
+  }
+
+  @Test
+  public void testPubSub() throws Exception {
+    final DaprRun daprRun = closeLater(startDaprApp(
+        this.getClass().getSimpleName(),
+        60000));
+
+    try (DaprClient client = daprRun.newDaprClient();
+         DaprPreviewClient previewClient = daprRun.newDaprPreviewClient()) {
+      for (int i = 0; i < NUM_MESSAGES; i++) {
+        String message = String.format("This is message #%d on topic %s", i, TOPIC_NAME);
+        //Publishing messages
+        client.publishEvent(PUBSUB_NAME, TOPIC_NAME, message).block();
+        System.out.println(
+            String.format("Published message: '%s' to topic '%s' pubsub_name '%s'", message, TOPIC_NAME, PUBSUB_NAME));
+      }
+
+      System.out.println("Starting subscription for " + TOPIC_NAME);
+
+      List<CloudEvent<String>> messages = Collections.synchronizedList(new ArrayList<>());
+      var listener = new SubscriptionListener<String>() {
+        @Override
+        public void onEvent(CloudEvent<String> event) {
+          messages.add(event);
+        }
+
+        @Override
+        public void onError(RuntimeException exception) {
+          System.out.println(exception);
+        }
+
+      };
+      try(var subscription = previewClient.subscribe(PUBSUB_NAME, TOPIC_NAME, listener, TypeRef.STRING)) {
+        callWithRetry(() -> {
```

Comment:
The grpc stream is being cancelled right away and no message is delivered beyond the first one. No idea why:

```
Published message: 'This is message #9 on topic stream-topic' to topic 'stream-topic' pubsub_name 'messagebus'
Starting subscription for stream-topic
Checking results for topic stream-topic
time="2024-09-11T16:43:39.455058-07:00" level=info msg="Subscribing to pubsub 'messagebus' topic 'stream-topic'" app_id=pubsubstreamit instance=Arturs-Laptop.local scope=dapr.runtime.pubsub.streamer type=log ver=1.14.2
time="2024-09-11T16:43:39.46075-07:00" level=info msg="Unsubscribed from pubsub 'messagebus' topic 'stream-topic'" app_id=pubsubstreamit instance=Arturs-Laptop.local scope=dapr.runtime.pubsub.streamer type=log ver=1.14.2
io.dapr.exceptions.DaprException: CANCELLED: Failed to read message.
Checking results for topic stream-topic
```